benches/network.rs

```diff
@@ -1,51 +1,86 @@
 // Copyright (c) Anza Technology, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use alpenglow::consensus::{ConsensusMessage, Vote};
+use std::mem::MaybeUninit;
+
+use alpenglow::consensus::{Cert, ConsensusMessage, NotarCert, Vote};
+use alpenglow::crypto::aggsig::SecretKey;
 use alpenglow::crypto::{Hash, aggsig, signature};
-use alpenglow::network::BINCODE_CONFIG;
+use alpenglow::network::localhost_ip_sockaddr;
 use alpenglow::shredder::{MAX_DATA_PER_SLICE, RegularShredder, Shred, Shredder};
 use alpenglow::types::Slot;
 use alpenglow::types::slice::create_slice_with_invalid_txs;
+use alpenglow::{ValidatorId, ValidatorInfo};
 use divan::counter::{BytesCount, ItemsCount};
-use rand::RngCore;
+use rand::prelude::*;
 
 fn main() {
     // run registered benchmarks.
     divan::main();
 }
 
+fn generate_vote() -> Vote {
+    let mut rng = rand::rng();
+    let mut hash = Hash::default();
+    rng.fill_bytes(&mut hash);
+    let sk = aggsig::SecretKey::new(&mut rng);
+    Vote::new_notar(Slot::new(0), hash.into(), &sk, 0)
+}
+
 #[divan::bench]
 fn serialize_vote(bencher: divan::Bencher) {
+    bencher
+        .counter(ItemsCount::new(1_usize))
+        .with_inputs(|| ConsensusMessage::Vote(generate_vote()))
+        .bench_values(|msg: ConsensusMessage| wincode::serialize(&msg).unwrap());
+}
+
+#[divan::bench]
+fn deserialize_vote(bencher: divan::Bencher) {
     bencher
         .counter(ItemsCount::new(1_usize))
         .with_inputs(|| {
-            let mut rng = rand::rng();
-            let mut hash = Hash::default();
-            rng.fill_bytes(&mut hash);
-            let sk = aggsig::SecretKey::new(&mut rng);
-            let vote = Vote::new_notar(Slot::new(0), hash.into(), &sk, 0);
-            ConsensusMessage::Vote(vote)
+            let msg = ConsensusMessage::Vote(generate_vote());
+            wincode::serialize(&msg).unwrap()
         })
-        .bench_values(|msg: ConsensusMessage| bincode::serde::encode_to_vec(msg, BINCODE_CONFIG));
+        .bench_values(|bytes: Vec<u8>| {
+            let _msg: ConsensusMessage = wincode::deserialize(&bytes).unwrap();
```

Comment:
does this deserialize the same vote again and again?  Are we worried about some weird caching effects giving a false impression that this is faster than it really is.